examples/federated-learning/tff/nvflare/README.md

```diff
@@ -0,0 +1,255 @@
+```
+
+Once the bucket has been generated, , modify the `pv-workspace.yaml.tpl` file to replace the bucket name:
+
+```bash
+export NVFLARE_WORKSPACE_BUCKET_NAME="\"$(terraform output -raw nvflare_workspace_bucket_name)\""
+envsubst < ../examples/federated-learning/tff/nvflare/templates/pv-workspace.yaml.tpl > ../examples/federated-learning/tff/nvflare/workload-pkg/base-storage/pv-workspace.yaml
+```
+
+Regenerate the `kustomization.yaml` file with all the necessary informations:
+
+```bash
+export NVFLARE_EXAMPLE_WORKLOADS_KUBERNETES_NAMESPACE="$(terraform output -raw nvflare_workloads_kubernetes_namespace)"
+envsubst < ../examples/federated-learning/tff/nvflare/templates/pv-workspace.yaml.tpl > ../examples/federated-learning/tff/nvflare/workload-pkg/base-storage/pv-workspace.yaml
+```
+
+### 3. Create Nvflare folder structure
+
+Now that the persistent volume is created, you can start creating the folder structure and upload it to Cloud Storage. First install nvflare on your workstation:
+
+```bash
+python3 -m pip install -r requirements.txt
+```
+
+Then, run the provisionning tool:
+
+```bash
+cd ${HOME}
+nvflare provision
+```
+
+You have to choose the non-ha (non high-availability) choice, as ha (high-availability) is not yet supported on Kubernetes. It will create a `project.yml` file that you can customize for your needs. Then, re-run the provisionning tool:
+
+```bash
+cd ${HOME}
+nvflare provision
+cd ${HOME}/workspace/example_project/prod_00/
+```
+
+The different folders generated represent the infrastructure you will deploy on the reference architecture:
+
+- `server1` is the server that will aggregate all the results from the computation
+- `site-1` and `site-2` are the clients that will be connected to the server
+- `admin@nvidia.com` is the administration client to start and list jobs
+
+### 4. Clone the repository
+
+You will need to clone the Nvflare repository that contains the job you will run on the reference architecture:
+
+```bash
+cd ${HOME}
+git clone --branch 2.4.2 https://github.com/NVIDIA/NVFlare
+```
+
+In the newly create folder, there is an `examples` folder that contains lots of demo you can use to test Nvflare. The TensorFlow demo is in the `examples/hello-world/hello-tf2`. The job will need to be in a special folder inside the `admin@nvidia.com` called the `transfer` folder to be deployed on each client participating in the computation.
+Copy this folder in the `transfer` folder of the `admin@nvidia.com` folder:
+
+```bash
+cd ${HOME}/workspace/example_project/prod_00/admin@nvidia.com/transfer
+cp -R ${HOME}/NVFlare/examples/hello-world/hello-tf2 .
+```
+
+Now, copy the whole workspace folder in Cloud Storage. The pods will have access to the infrastructure to run jobs:
+
+```bash
+gcloud storage -m cp -r ${HOME}/workspace gs://${NVFLARE_WORKSPACE_BUCKET_NAME}
+```
+
+### 5. Deploy the infrastructure
+
+Everything is now setup to be able to submit the job. Deploy the infrastructure via ConfigSync:
+
+```bash
```

Comment:
Fix the destination paths in this snippet

Reply:
Note that https://github.com/GoogleCloudPlatform/federated-learning/pull/485 adds the necessary terraform outputs to read them from Terraform. But we don't likely want to be blocked by that.